src/components/auth/registerUser.functional.spec.js

```diff
@@ -0,0 +1,139 @@
+
+    makeTemplate = angular.element(signUpTemplate);
+
+    $compile(makeTemplate)($rootScope);
+
+    form = $rootScope.signup.form;
+
+    makeController = () => {
+      return new signUpCtrl($auth, $state, $toast, AuthService);
+    };
+  }));
+  it('signup form test', function() {
+    const controller = makeController();    
+
+    form.email.$setViewValue('eeeiii');
+    $rootScope.$digest();
+    expect(form.email.$valid).to.eq(false);
+    expect(form.email.$viewValue).to.eq('eeeiii');
+    expect(form.$invalid).to.eq(true);
+
+    form.email.$setViewValue('');
+    $rootScope.$digest();
+    expect(form.email.$valid).to.eq(false);
+    expect(form.email.$viewValue).to.eq('');
+    expect(form.$invalid).to.eq(true);
+
+    form.email.$setViewValue('chaoeninwinstack.com');
+    $rootScope.$digest();
+    expect(form.email.$valid).to.eq(false);
+    expect(form.email.$viewValue).to.eq('chaoeninwinstack.com');
+    expect(form.$invalid).to.eq(true);
+
+    form.email.$setViewValue('chaoen@inwinstack.com');
+    $rootScope.$digest();
+    expect(form.email.$valid).to.eq(true);
+    expect(form.email.$viewValue).to.eq('chaoen@inwinstack.com');
+    expect(form.$invalid).to.eq(true);
+
+    const data = { email : form.email.$setViewValue };
+    controller.form = { email: { '$valid': form.email.$valid }};
+    controller.credentials = data;
+    $httpBackend.expectPOST('http://163.17.136.83:8080/api/v1/auth/checkEmail', data).respond(403);
+    controller.checkEmail();
+    $httpBackend.flush();
+    $rootScope.$digest();
+    expect(controller.emailIsValid).to.eq(false);
+
+    form.email.$setViewValue('chaoen.l@inwinstack.com');
+    $rootScope.$digest();
+    expect(form.email.$valid).to.eq(true);
+    expect(form.email.$viewValue).to.eq('chaoen.l@inwinstack.com');
+    expect(form.$invalid).to.eq(true);
+    
+    data.email = form.email.$setViewValue;
+    controller.credentials = data;
+    $httpBackend.expectPOST('http://163.17.136.83:8080/api/v1/auth/checkEmail', data).respond(200);
+    controller.checkEmail();
+    $httpBackend.flush();
+    $rootScope.$digest();
+    expect(controller.emailIsValid).to.eq(true);
+
+    form.password.$setViewValue('abc1234');
+    form.password_confirmation.$setViewValue('1234567');
+    $rootScope.$digest();
+    expect(form.password_confirmation.$valid).to.eq(false);
+    expect(form.password_confirmation.$viewValue).to.eq('1234567');
+    expect(form.password.$viewValue).to.eq('abc1234');
+    expect(form.$invalid).to.eq(true);
+    
+    form.password_confirmation.$setViewValue('abc1234');
+    $rootScope.$digest();
+    expect(form.password_confirmation.$valid).to.eq(true);
+    expect(form.password_confirmation.$viewValue).to.eq('abc1234');
+    expect(form.password.$viewValue).to.eq('abc1234');
+    expect(form.password.$valid).to.eq(true);
+    expect(form.email.$valid).to.eq(true);
+    expect(form.$invalid).to.eq(false);
+    expect(controller.emailIsInvalid).to.eq(false);
```

Comment:
Line 97 to 103 and line 105 to 113 should be separated to different it() sections.